src/webgpu/api/validation/resource_usages/textureUsageInPassEncoder.spec.ts

```diff
@@ -124,33 +127,74 @@ class TextureUsageTracking extends ValidationTest {
     });
   }
 
-  testValidationScope(): {
+  testValidationScope(
+    compute: boolean
+  ): {
     bindGroup0: GPUBindGroup;
     bindGroup1: GPUBindGroup;
     encoder: GPUCommandEncoder;
-    pass: GPURenderPassEncoder;
-    pipeline: GPURenderPipeline;
+    renderPass: GPURenderPassEncoder;
+    computePass: GPUComputePassEncoder;
+    renderPipeline: GPURenderPipeline;
+    computePipeline: GPUComputePipeline;
```

Comment:
Can this possibly be made to return:
```suggestion
    pass: GPURenderPassEncoder | GPUComputePassEncoder;
    pipeline: GPURenderPipeline | GPUComputePipeline;
```
instead?

Reply:
Some APIs are exposed in GPUProgrammablePassEncoder, for example, setBindGroup. And endPass() exists in both GPURenderPassEncoder and GPUComputePassEncoder. These APIs can be shared between GPURenderPassEncoder and GPUComputePassEncoder. We can reuse the code for both render and compute easily. However, some APIs belong to a specific encoder, for example, setPipeline (both encoders have setPipeline, but these two setPipeline() take different parameters), draw or dispatch, etc. So I need to return renderPass/computePass, renderPipeline/computePipeline. Otherwise, the tests fail when I run "grunt pre" because of type mismatch. For example, it complained that "draw call requires a GPURenderPassEncoder, while you provide GPURenderPassEncoder | GPUComputePassEncoder."

Reply:
You can `if (pass instanceof GPURenderPassEncoder)` or `assert(pass instanceof GPURenderPassEncoder);` if needed. You can also use unsafe type casts (`pass as GPURenderPassEncoder`) if you can't get the type system to do what you want.

Reply:
Yeah. unsafe type cast is a good solution that can resolve the problem of sharing code between render pass and compute pass I met here.